tempto-kafka/src/main/java/io/prestodb/tempto/fulfillment/table/kafka/KafkaTableManager.java

```diff
@@ -0,0 +1,225 @@
+        if ((long) queryResult.row(0).get(0) != 1) {
+            throw new RuntimeException(format("Table %s.%s not defined if kafka catalog (%s)", schema, name, prestoKafkaCatalog));
+        }
+    }
+
+    private void deleteTopic(String topic)
+    {
+        withZookeeper(zkUtils -> {
+            if (AdminUtils.topicExists(zkUtils, topic)) {
+                AdminUtils.deleteTopic(zkUtils, topic);
+
+                for (int checkTry = 0; checkTry < 5; ++checkTry) {
+                    if (!AdminUtils.topicExists(zkUtils, topic)) {
+                        return;
+                    }
+                    try {
+                        Thread.sleep(1_000);
+                    }
+                    catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        throw new RuntimeException("could not delete topic " + topic);
+                    }
+                }
+                throw new RuntimeException("could not delete topic " + topic);
+            }
+        });
+    }
+
+    private void createTopic(String topic, int partitionsCount, int replicationLevel)
+    {
+        withZookeeper(zkUtils -> {
+            Properties topicConfiguration = new Properties();
+            AdminUtils.createTopic(zkUtils, topic, partitionsCount, replicationLevel, topicConfiguration, RackAwareMode.Disabled$.MODULE$);
+        });
+    }
+
+    private void insertDataIntoTopic(String topic, KafkaDataSource dataSource)
+    {
+        // create instance for properties to access producer configs
+        Properties props = new Properties();
+
+        props.put("bootstrap.servers", brokerHost + ":" + brokerPort);
+        props.put("acks", "all");
+        props.put("retries", 0);
+        props.put("key.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer");
+        props.put("value.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer");
+
+        Producer<byte[], byte[]> producer = new KafkaProducer<>(props);
+
+        Iterator<KafkaMessage> messages = dataSource.getMessages();
+        while (messages.hasNext()) {
+            KafkaMessage message = messages.next();
+            try {
+                producer.send(new ProducerRecord<>(topic,
+                        message.getPartition().isPresent() ? message.getPartition().getAsInt() : null,
+                        message.getKey().orElse(null),
+                        message.getValue())).get();
+            }
+            catch (Exception e) {
+                throw new RuntimeException("could not send message to topic " + topic);
+            }
+        }
+    }
+
+    private void withZookeeper(Consumer<ZkUtils> routine)
+    {
+        int sessionTimeOutInMs = 15_000;
+        int connectionTimeOutInMs = 10_000;
+        String zookeeperHosts = zookeeperHost + ":" + zookeeperPort;
+
+        ZkClient zkClient = new ZkClient(zookeeperHosts,
```

Comment:
is it a good idea to create a client per each request?

Reply:
There are not to many of those. A few per topic. I would not make change it unless we see this is a problem.